### PR TITLE
[SW-2191] Distribute Mojos via SparkFiles to Avoid Maximum Array Size Limit

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: ml/src/main/scala/ai/h2o/sparkling/ml/models/H2OTargetEncoderModel.scala [hunk 1]
```diff
@@ -38,9 +38,9 @@ class H2OTargetEncoderModel(override val uid: String, targetEncoderModel: H2OMod
   with RestCommunication {
 
   lazy val mojoModel: H2OTargetEncoderMOJOModel = {
-    val mojoData = targetEncoderModel.downloadMojoData()
+    val mojo = targetEncoderModel.downloadMojo()
     val model = new H2OTargetEncoderMOJOModel()
-    copyValues(model).setMojoData(mojoData)
+    copyValues(model).setMojoData(mojo)
```
Comment (jakubhava) [inline ml/src/main/scala/ai/h2o/sparkling/ml/models/H2OTargetEncoderModel.scala:43]: The name of this method should be changed as well to reflect that we are using MOJO, not mojo data

File: scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModel.scala [hunk 2]
```diff
@@ -261,15 +266,14 @@ object H2OMOJOModel extends H2OMOJOReadable[H2OMOJOModel] with H2OMOJOLoader[H2O
...
-    model.setMojoData(mojoData)
     model
   }
 
   // Internal method used only within Sparkling Water pipelines.
   // When H2OMOJOModel is created from existing mojo created in H2O-3, we set features names as features stored in mojo
   // (they are not nested and structured), but as in Spark, data frames can be nested, we need to handle it
   private[h2o] def createFromMojo(
-      mojoData: Array[Byte],
+      mojoData: File,
```
Comment (jakubhava) [inline scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModel.scala:276]: All fields and methods referencing mojoData should be updated to a new name

File: scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOBaseCache.scala [hunk 3]
```diff
@@ -71,13 +73,13 @@ trait H2OMOJOBaseCache[B, M] extends Logging {
     }
   }
 
-  def getMojoBackend(uid: String, bytesGetter: () => Array[Byte], model: M): B = Lock.synchronized {
+  def getMojoBackend(uid: String, mojoDataGetter: () => File, model: M): B = Lock.synchronized {
```
Comment (jakubhava) [inline scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOBaseCache.scala:76]: `mojoGetter` ?

File: scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOBaseCache.scala [hunk 4]
```diff
@@ -71,13 +73,13 @@ trait H2OMOJOBaseCache[B, M] extends Logging {
...
     if (!pipelineCache.contains(uid)) {
-      pipelineCache.put(uid, loadMojoBackend(bytesGetter(), model))
+      pipelineCache.put(uid, loadMojoBackend(mojoDataGetter(), model))
     }
     lastAccessMap.put(uid, System.currentTimeMillis())
     pipelineCache(uid)
   }
 
-  def loadMojoBackend(mojoData: Array[Byte], model: M): B
+  def loadMojoBackend(mojoData: File, model: M): B
```
Comment (jakubhava) [inline scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOBaseCache.scala:84]: just `mojo` or `mojoFile` ?

**Comment by mn-mikke** [comment]
Ok, will rename. Thanks!

**Pull request merged**